### PR TITLE
Document inherited methods

### LINKED ISSUE #1125: Showing me a type without the [inherited] methods defined on it is pretty useless
Trying to use Hacklang for Advent of code.

A documentation file like this: https://docs.hhvm.com/hsl/reference/interface/HH.Lib.File.CloseableReadHandle/

Is basically useless? Maybe I'm not sure how to use this website correctly... but I want to see the methods defined on a type. I'm trying to read in a text file, and every thing just links to interfaces like this with no methods defined. Even if they're not *defined* on this type, and come from a class that this type extended or an interface it implemented, I think this documentation would only be useful if I could see those methods here.

### PR DESCRIPTION
**Pull request by fredemmott**
This (more than slightly) needs a refactoring cleanup... will do that in
a followup, then look at the type/newtype stuff (which is currently not
supported at all, and has been requested for a while)

fixes #1125

### CODE REVIEW

**Comment by fredemmott** [comment]
First commit is #1163

Second commit gets me this (ignoring the squished layout because of minimum browser zoom):


image

File: src/build/HHAPIDocBuildStep.php [hunk 1]
```diff
@@ -107,14 +101,46 @@ public function buildAll(): void {
...
+      }
+      if ($def is ScannedInterface) {
+        $hh_apidoc_index['interfaces'][$def->getName()] = $documentable;
+        continue;
+      }
+      if ($def is ScannedTrait) {
+        $hh_apidoc_index['traits'][$def->getName()] = $documentable;
+        continue;
+      }
+    }
```
Comment (Atry) [inline src/build/HHAPIDocBuildStep.php:136]: I would use `Dict\group_by` to avoid mixing up of functional programming style code and the mutable loop.

Comment (fredemmott) [reply src/build/HHAPIDocBuildStep.php:136]: `Dict\group_by()` doesn't produce shapes, and would need a call to reflection (or `get_class()` to significantly clean this up

Comment (fredemmott) [reply src/build/HHAPIDocBuildStep.php:136]: There also currently isn't functional style here?

**Review by Atry — APPROVED** [review]

File: src/build/HHAPIDocBuildStep.php (hunk 1 above)
Comment (Atry) [reply src/build/HHAPIDocBuildStep.php:136]: I think it would be good if we have a high level API in `Facebook\HHAPIDoc`

Comment (fredemmott) [reply src/build/HHAPIDocBuildStep.php:136]: HHAPIDoc does have one; docs.hhvm.com uses its' own as there's multiple sources (hhvm, HHI, hsl, hsl-experimental) and they can contain contradictory definitions (hhvm + hhi) which need resolving.

**Pull request merged**